src/thruster_types/thruster_types.proto

```diff
@@ -1,6 +1,19 @@
 package rock_thruster.msgs;
 
-message ThrusterInput
+message Raw
 {
-    optional double     frequency = 1 [default = 0];
+    optional string         name = 1;
```

Comment:
> Some people advise to use only optional and repeated fields.

This is for super-long-lived client/server protocols where upgrading the protocol is a necessity. I don't believe we have such requirements.

However, if you go for the single structure (which I would advise), then you'll need to mark the fields as optional anyways ;-)

> If I create only the Thruster msg I think I would need a "mode" attribute (am I wrong?)

I don't think so. It seems that you can test whether optional fields are present or not with the has_*() methods.